### DISCUSSION TITLE
I get sometimes many reconnection in the iot device side

### DISCUSSION THREAD

**Discussion by roysG** [emqx/emqx#10934] [Q&A]
Hi,
I get sometimes many reconnections in the iot device side,
I want to check if the problem is in our cluster emqx or maybe in the side of the network company.

Can someone tell me please how to observe the logs of the emqx and what should i need to look for..?

I am using today in version 4.3.5
Thanks

**Comment by lafirest** [comment]
@roysG 
Version 4.3.5 is out of maintenance, please have a try with a newer version.
For observing, you can set the log level to debug, at this level, EMQX will print the `connect` and `disconnect` and the `socket closed` events, I think these could help you to find the reconnection reason.
I guess the reason may be that your devices do not periodically send the heartbeat packet.

**Comment by roysG** [comment]
Hi,
Thanks for you shortly response,
Today i use in library called TinyGSM, i am sure you know or hear about it
https://github.com/vshymanskyy/TinyGSM

I defined the parameter mqttClient.setKeepAlive to 15 sec:
mqttClient.setKeepAlive(15);

Do you think maybe is the reason or maybe other thinking i need to check?

Thanks.

**Reply by lafirest** [reply]
I am sorry that I have never used this library. In my experience that this line may only set the keepalive parameter for connection, you still need to send the MQTT heartbeat packet periodically.
To find the real reason, setting the debug level to `debug` is useful for us to see why or we can only blind guess.
